collector/perfos.go

```diff
@@ -0,0 +1,133 @@
+// returns data points from Win32_PerfRawData_PerfOS_System class (undocumented)
+
+package collector
+
+import (
+	"log"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Factories["perfos"] = NewPerfOSCollector
+}
+
+// A PerfOSCollector is a Prometheus collector for WMI metrics
+type PerfOSCollector struct {
+	// XXX names
+	ContextSwitchesPersec     *prometheus.Desc
```

Comment:
Not sure at all (and there's no official documentation of these fields)
Here's an example output:

```
# TYPE wmi_perfos_context_switches_per_sec gauge
wmi_perfos_context_switches_per_sec 4.272205793e+09
# TYPE wmi_perfos_exception_dispatches_per_sec gauge
wmi_perfos_exception_dispatches_per_sec 1.2127682e+07
# TYPE wmi_perfos_system_calls_per_sec gauge
wmi_perfos_system_calls_per_sec 4.26178136e+08
```